types/ws/ws-tests.ts

```diff
@@ -38,16 +41,16 @@ var WebSocketServer = WebSocket.Server;
 }
 
 {
-    var wsc = new WebSocket('ws://echo.websocket.org/', {
-        protocolVersion: 8,
-        origin: 'http://websocket.org'
-    });
+    var wsc = new WebSocket('ws://echo.websocket.org/');
 
     wsc.on('open',  () => wsc.send(Date.now().toString(), {mask: true}));
     wsc.on('close', () => console.log('disconnected'));
+    wsc.on('error', (error) => {
+        console.log(`unexpected response: ${error}`);
+    });
 
-    wsc.on('message', (data, flags) => {
-        console.log('Roundtrip time: ' + (Date.now() - parseInt(data)) + 'ms', flags);
+    wsc.on('message', (data: any) => {
```

Comment:
Correct: should be explicit here.

Package users can alternatively specify the union types—`wsc.on('message', (data: string|Buffer|ArrayBuffer|Buffer[]) => { ... }`—and then `typeof message` in a switch/if-else to handle each possible type as (if) needed.